### ISSUE TITLE
Remove/delete room

### ISSUE THREAD

**Issue by rgpublic** [vector-im/element-meta#466]

**Comment by hex-m** [comment]
As a user my expectation of "remove room" would be just that: the room should not exist anymore. (not having any members, not be join-able by anyone, deleted after some grace-period)

> often the user is just trying to recover the alias

The management of aliases (actually called Addresses in the UI) is a different issue imo and I would not click a "delete room" button when I search for that feature.

I would expect a workflow like this:
1. click "delete room"
2. if there are other admins get message: "ask all other administrators to demote themselves or leave the room first"
3. "Warning: you and all other members will loose access to this room and its content!"
4. "Should the Room Address '#room1234:server.tld' be pointed to a different room? Should the current members be invited to join that room?"
5. all aliases/addresses get deleted, all members kicked, history visibility is set to members only, room is set to invite-only and garbage collected sooner or later.

> If you're the server admin [..]

Should Element-Web really include administration-functionality? There are issues (https://github.com/vector-im/element-meta/issues/1237, https://github.com/matrix-org/synapse/issues/2032) requesting a dedicated interface for such things.

**Comment by hex-m** [comment]
Even using the [Admin API](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/rooms.md#delete-room-api) to delete a room turns up odd behaviour. It may be useful to track that as a separate issue from a room admin trying to delete the room but I'll put this here for now.
* I deleted the room on the server it was created (according to the `room_id`) and would expect that this would kick all users from the room - not only the ones that are on my server.
* Users on servers don't get any notification except for "User X left the room". They have no idea the room was "deleted".
* The local (kicked) user still has the room in the "Historical" section, and when she tries to join again gets the message: "No known servers". This feels extra wrong as I didn't set the `block` flag at the delete-call.
* A federated user left in the room can re-invite the kicked users and thus undo the "delete"-operation. After re-joining the user has the power level she had before.

Suggestions that would be possible without spec change:
* send a message into the room explaining what is happening
* set a reason explaining why you were kicked
* element-web: allow access to the history of rooms you were kicked from. https://github.com/vector-im/element-meta/issues/465
* if all admins are on my server: kick **all** users from the room (including federated ones). Otherwise warn that the room can not be deleted until the federated admins leave.

Possible spec changes:
* new room state `deletion_proposed` that informs all admins about the wish to delete the room. either server-admins or room-admins can `confirm`. If all users/servers confirmed the room gets deleted.
* new room state `deleted` that tells all servers to remove their users and forgets about the room. (maybe keeping it read-only for some grace period)
* as having more or less than one admin is a problem for many things, define a unique owner/creator-role for a room. https://github.com/matrix-org/matrix-doc/issues/652

**Comment by shmerl** [comment]
Use case: 1:1 room, one user got locked out from the account, so they can't leave the room. There should be an ability to remove such room by the other user (who is also admin).

**Comment by tanriol** [comment]
If you mean that in an 1:1 room any user should be able to delete the room without any traces, I'd be opposed to that. As a user, I don't want the person I'm chatting with to be able to completely remove our communication history.

**Comment by shmerl** [comment]
It's kind of already possible for individual messages, including "Remove recent messages" for more bulk deletions. So not conceptually very different, just a bigger scale.

**Comment by movahhedi** [comment]
Still waiting for an update on this. It's very annoying

**Comment by BMaxV** [comment]
Got here from trying to leave a test room and making it disappear.

Turns out you can't. That's pretty funny.

**Comment by DantesNoWar** [comment]
Hi, I just tried to create a Space test and to build a space inside the other to three levels. I have created also sone test room inside of them. Now, how can I delete those tests spaces and rooms? 
I don't want to leave this garbage alive.
Is there some news?

**Comment by spaetz** [comment]
If you leave rooms and spaces, they will be cleaned up according to server policy. If you set these rooms as "invite only", noone will ever be able to enter them again (including you) if everyone has left the rooms until they are deleted by the server....

**Comment by rgpublic** [comment]
@spaetz  But when does the server delete those rooms? If we are talking about Matrix Synapse I don't know about a room retention policy. There's a message retention policy, but not a room retention policy. All in all, I still think the room management is really confusing. I'm checking e.g. my "rooms" table in the database on the server and there are hundreds of rooms listed though we are currently only using a handful. The database gets polluted with all that stuff and there still isn't a real clear solution. But OTOH, this probably much rather a Synapse issue than an Element issue.

**Comment by markcst** [comment]
Hi, is this still a thing? I know right we could leave the room and wait for the server to delete it, but that's not the same thing as directly delete it.

This shouldn't be a thing in almost 2026 for a chat service, please just work on this feature asap

**Comment by markcst** [comment]
> It's weird for end-users not to be able to "delete" a room: it does not really matter if the room is left abandonned, but it would make sense for admins to have a "delete" button that kicks everyone, lock and leave. Homeserver GC can take care of the rest, but I think it doesn't matter from the user point of view.
> 
> It should not be hard to implement in riot.

I completely agree. I think are things like these that push more users away from wonderful protocols and services like Matrix/Element. I hope devs take it in consideration

**Comment by DantesNoWar** [comment]
@markcst I agree. But I don't know if these suggestions are taken in consideration or not.
Anyway, this issue should be absolutely fixed ASAP with a contextual removal request by the admin of the room/space.
The removal request should be taken in serious consideration by the servers and executed ASAP. This is more a politic rather than a complete feature but it will improve the admins experience.

**Comment by spaetz** [comment]
You are confounding two things: db tables and space, and what is visible from a room/space admin perspective.

1) if you are talking about a room admin/space admin perspective, just have every one leave the room, remove the aliases and forget it. 1b) use corresponding bots and tooling to make this easy (meowlnir or other bots). 2) If you are a server admin, use the purge room API, or a CLI tool to achieve the same with a single command.
All that is unrelated whether the empty room shell will remain in the database.
3) synapse is bad at deleting forgotten, empty rooms from its db (and that is a small issue) but that is usually just a minor space issue (unless you are creating and leaving thousands of rooms) which is not visible to room or server admin unless you manually poke in the db. The purge API exists and is usable and will delete these from the db.

All if this is not an element-meta issue or spec issue, but squarely a synapse issue, if at all.

Over and out.

**Comment by DantesNoWar** [comment]
I understand the technical distinction @spaetz , but I believe you're underestimating the user experience issue.

The fact is:

1.  Normal users shouldn’t need bots or scripts to delete a room they created.
2.  "Leave and forget" isn’t equivalent to "delete". Abandoning a room leaves it potentially visible and reusable, compromising privacy and control.
3.  The room’s persistence after abandonment remains a concern for users.
4.  Claiming it's solely a Synapse "minor" issue deflects responsibility; Element, as the reference client, should drive ecosystem improvements.

Several solutions have been proposed, including an "Abandon room" button (@Fnux ), a PHP script workaround (@karl007 ), and @mahemoff's suggestion – fading out the delete button and requiring a single administrator – to address the complexities of multi-admin rooms. While the functionality to "leave a room" exists in Element, it doesn't provide the definitive removal desired by us users.

There's a disconnection between technical capabilities and user-centered design that needs to be addressed, especially considering this issue has been open for 7 years by now.

**Comment by pH142857** [comment]
I fully agree with @DantesNoWar . This is one of numerous (tons of) examples of libre software builders underestimating the UX standpoint about a fundamental feature. And my experience strongly suggests that this is one of the main reasons people struggle to switch to libre software.

I'm unsure about Element's business model and team, but does it include designers? What do they say about such things?

**Comment by rgpublic** [comment]
I just tried to find out what the current state is. It seems in fact Matrix Synapse has this implemented in the meantime:

https://matrix-org.github.io/synapse/v1.40/admin_api/rooms.html#delete-room-api

Quote: "This API will remove all trace of the old room from your database after removing all local users. If purge is true (the default), all traces of the old room will be removed from your database after removing all local users."

The only thing that should IMHO be done is having this available in the room context menu of the Element Client. IMHO this isn't something that should only be available in the admin UI. If users can create new rooms from the client, they should be able to get rid of them.

**Comment by markcst** [comment]
@rgpublic 
> I just tried to find out what the current state is. It seems in fact Matrix Synapse has this implemented in the meantime:
> 
> https://matrix-org.github.io/synapse/v1.40/admin_api/rooms.html#delete-room-api
> 
> Quote: "This API will remove all trace of the old room from your database after removing all local users. If purge is true (the default), all traces of the old room will be removed from your database after removing all local users."

Nice finding!

> The only thing that should IMHO be done is having this available in the room context menu of the Element Client. IMHO this isn't something that should only be available in the admin UI. If users can create new rooms from the client, they should be able to get rid of them.

I absolutely agree 💯

**Comment by t3chguy** [comment]
> The only thing that should IMHO be done is having this available in the room context menu of the Element Client. IMHO this isn't something that should only be available in the admin UI. If users can create new rooms from the client, they should be able to get rid of them.



It'd need to be a Matrix API, not a Synapse API, for that to be considered.

**Comment by rgpublic** [comment]
Understood. Thanks for the clarification. So guess we are here:

https://github.com/matrix-org/matrix-spec/issues/431

Which is still open. Sigh. This is beyond frustrating and IMHO also actually a tad bit ridiculous. We have a wonderful Chat system. Everything works, syncs, you have all sorts of wonderful special features, everyone loves it, but a user can't delete a room s/he has just created a second ago. Sometimes, I just don't understand this world ;-/

**Comment by DantesNoWar** [comment]
@rgpublic I guess we have to wait for a new fork, contribute to an existing one, or even create a new one. Element seems stuck with old unresolved issues on different areas that will not be fixed for many years.

Meanwhile, new private and decentralized FOSS competitors are shortening the technological and user gap with Element and the Matrix network, and they seem to listen more to their own community.

Coincidences?

We, as community owners, contributors and users, want a polished and private chat and group experience that just works fine by maintaining its private-by-design roots.
Likewise, we can't rely for long on a project that can't fix its structural problems, even if we all love FOSS.

What else can we do if our feedbacks aren't listened?

We just want to be able to talk to the dev team and make our voices heard, not to wait more than 7 years just to be told “it's not an Element fault” and then complete silence for more than a month. Where are the team?

Is this even a community-driven FOSS project or not?

The community is watching. If Element wants to stay relevant, it needs to listen—through better UX, communication, and a stronger commitment to its users. Otherwise, we’ll go where we’re heard.

Remember that Element and Matrix are one. The same people who build the protocol also build the client.
If the team doesn’t hear the community, it’s not just a mere protocol issue—it’s something else…
Pointing fingers isn’t professional, and it doesn’t help the project nor the users.
<img width="1910" height="1018" alt="Image" src="https://github.com/user-attachments/assets/3b382743-2cd4-4374-97f1-07f92a7a9d44" />
